src/core/formatters.ts

```diff
@@ -131,6 +131,56 @@ export module Formatters {
   }
 
   /**
+   * Creates a formatter for values that displays abbreviated values
+   * and uses standard short scale suffixes (thousands, millions, billions, trillions,
+   * quadrillions).
+   *
+   * Numbers with a magnitude outside of (10 ^ (-precision), 10 ^ (15 + precision)) are shown using
+   * scientific notation to avoid creating extremely long decimal strings.  The inputs to the
+   * formatter are primarily expected to be ranges of common currency values.
+   *
+   * @param {number} [precision] the number of decimal places to show (default 3)
+   *
+   * @returns {Formatter} A formatter for large numbers.
+   */
+  export function shortScale(precision = 3) {
+    verifyPrecision(precision);
+    var suffixes = "KMBTQ";
+    var eFormatter = d3.format("." + precision + "e");
+    var fFormatter = d3.format("." + precision + "f");
+    var max = Math.pow(10, (3 * (suffixes.length + 1)));
+    var min = Math.pow(10, -precision);
+    return (num: number) => {
+      var absNum = Math.abs(num);
+      if (absNum < min || absNum >= max) {
+        return eFormatter(num);
+      }
+      var factor = 1;
+      var idx = -1;
+      while (absNum >= (1000 * factor) && idx < (suffixes.length - 1)) {
+        idx += 1;
+        factor *= 1000;
+      }
+      var output = "";
+      if (idx === -1) {
+        output = fFormatter(num);
+      } else {
+        output = fFormatter(num / factor) + suffixes[idx];
+      }
+      // catch rounding by the underlying d3 formatter
+      if ((num > 0 && output.substr(0, 5) === "1000.") || (num < 0 && output.substr(0, 6) === "-1000.")) {
+        if (idx < suffixes.length - 1) {
+          factor *= 1000;
+          idx += 1;
+          output = fFormatter(num / factor) + suffixes[idx];
```

Comment:
It seems that `factor` and `idx` seem to share some meaning together.  Instead of having 2 variables, why can't we have one variable (maybe `idx`) and then derive the other number where needed (factor seems to be consistently `Math.pow(1000, idx + 1)`

A lot of this code from my perspective is pretty confusing and having 2 variables that share similar information is part of the reason.